CRM/Upgrade/Incremental/php/FiveSixtyOne.php

```diff
@@ -21,6 +21,23 @@
  */
 class CRM_Upgrade_Incremental_php_FiveSixtyOne extends CRM_Upgrade_Incremental_Base {
 
+  public function setPreUpgradeMessage(&$preUpgradeMessage, $rev, $currentVer = NULL): void {
+    if ($rev === '5.61.alpha1' && CRM_Core_DAO::singleValueQuery('SELECT id FROM civicrm_contribution_recur LIMIT 1')) {
+      $documentationUrl = 'https://docs.civicrm.org/dev/en/latest/financial/recurring-contributions/';
+      $documentationAnchor = 'target="_blank" href="' . htmlentities($documentationUrl) . '"';
+      $extensionUrl = 'https://docs.civicrm.org/dev/en/latest/financial/recurring-contributions/';
+      $extensionAnchor = 'target="_blank" href="' . htmlentities($extensionUrl) . '"';
+
+      $preUpgradeMessage .= '<p>' .
+        ts('This release contains a change to the behaviour of recurring contributions under some edge-case circumstances.')
+        . ' ' . ts('In recent releases the amount and currency on the recurring contribution record changed when the amount of any contribution against it was changed, indicating a change in future intent.')
```

Comment:
```suggestion
        . ' ' . ts('Since 5.49 the amount and currency on the recurring contribution record changed when the amount of any contribution against it was changed, indicating a change in future intent.')
```